src/figure_finder.py

```diff
@@ -0,0 +1,185 @@
+                        print(f"Bbox area to frame ratio too small: {bbox_to_image_ratio}")
+                        continue
+
+                color = None
+                determined_type = None
+
+                if bbox_labeled.label in config["bbox_shrink"]["labels"]:
+                    shrink_bbox(bbox_labeled, 0.5)
+
+                if bbox_labeled.label in config["ball_color_verification"]["labels"]:
+                    figure_img = get_img_piece(frame, bbox_labeled)
+                    dominant_colors = self.color_detection.get_dominant_colors(figure_img, 2, show=False)
+                    color = self.color_detection.get_matching_color(dominant_colors)
+                    determined_type = config["ball_color_verification"]["color_mapping"].get(color, "unknown")
+
+                if bbox_labeled.label in config["label_mapping"]["labels"]:
+                    determined_type = config["label_mapping"]["mapping"].get(bbox_labeled.label, "unknown")
+
+                local_position = self.bbox_to_ground_position(bbox_labeled)
+                figure = Figure(bbox_labeled.label, bbox_labeled, local_frame_coords=local_position, color=color, determined_type=determined_type)
+                # print(f"Figure: {figure}")
+                figures.append(figure)
+            except Exception as e:
+                rospy.logwarn(f"Figure creation exception: {e}")
+
+        return figures
+
+    def publish_confirmed_figures(self, confirmed_figures: List[Figure]):
+        confirmed_msg_list = [f.to_msg(status="on_ground") for f in confirmed_figures]
+        figure_msg_list = FigureMsgList()
+        figure_msg_list.figures = confirmed_msg_list
+        self.confirmed_figures_pub.publish(figure_msg_list)
+
+    def detection_callback(self, image, bboxes_msg: BoundingBoxLabeledList):
+        frame = self.bridge.imgmsg_to_cv2(image, "bgr8")
+        figures = self.create_figures(frame, bboxes_msg.boxes, self.figure_operations_config)
+
+        self.figure_colletor.update(figures)
+        confirmed_figures = self.figure_colletor.confirm_figures()
+        if confirmed_figures:
+            print("Confirmed figures: ", confirmed_figures)
+            self.publish_confirmed_figures(confirmed_figures)
+
+    def global_pos_callback(self, msg):
+        self.last_telem["latitude"] = msg.latitude
+        self.last_telem["longitude"] = msg.longitude
+        self.last_telem["altitude_amsl"] = msg.altitude
+
+        rospy.loginfo_throttle(
+            10,
+            f"Received global position: Latitude: {msg.latitude}, Longitude: {msg.longitude}, Altitude(AMSL): {msg.altitude}",
+        )
+
+    def compass_callback(self, msg):
+        self.last_telem["heading"] = msg.data
+
+    def rel_alt_callback(self, msg):
+        self.last_alt_agl = msg.data
+        self.last_telem["altitude"] = msg.data
+        rospy.loginfo_throttle(10, f"Received relative Altitude(AGL): {msg.data}")
```

Comment:
Create telemdata dataclass to gather this information, because it will be easier to initialize